src/main/java/uk/gov/pay/webhooks/webhook/resource/WebhookResource.java

```diff
@@ -103,7 +107,39 @@ public SigningKeyResponse regenerateSigningKey(@PathParam("externalId") @NotNull
                     .map(WebhookResponse::from)
                     .toList();
     }
-    
+
+    @UnitOfWork
+    @Path("/{externalId}/messages")
+    @GET
+    public WebhookMessageSearchResponse getWebhookMessages(
+            @PathParam("externalId") String externalId,
+            @QueryParam("page") Integer page,
+            @QueryParam("status") String status
+    ) {
+        var currentPage = (page != null) ? page : 1;
```

Comment:
```suggestion
        var currentPage = Optional.ofNullable(page).orElse(1);
```
maybe a bit nicer? But only a style thing!

Reply:
Much prefer this, thanks 👍 including the above suggestion, I've gone with 

```java
var currentPage = Optional.ofNullable(page)
                .orElse(1);

 var deliveryStatus = Optional.ofNullable(status)
                .map(String::valueOf)
                .orElse(null);
```

I was out here trying to write functional JS 😅 
```javascript
const currentPage = page && 1
```